workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/AsDialogHolderWithContent.kt

```diff
@@ -22,15 +24,22 @@ import com.squareup.workflow1.ui.toViewFactory
  * [View.backPressedHandler][com.squareup.workflow1.ui.backPressedHandler],
  * and honor the [OverlayArea] and [CoveredByModal] values placed in
  * the [ViewEnvironment] by the standard [BodyAndOverlaysScreen] container.
+ *
+ * @param turnOffChrome by default, we tweak the window's background and flags
+ * to ensure it will honor [OverlayArea]. Disable this for complete control,
+ * with one exception: this method always adds the [FLAG_NOT_TOUCH_MODAL] flag, to
+ * ensure that events are blocked only within the expected bounds.
  */
 @WorkflowUiExperimentalApi
-public fun <C : Screen, O : ScreenOverlay<C>> ComponentDialog.setContent(
+public fun <C : Screen, O : ScreenOverlay<C>> ComponentDialog.asDialogHolderWithContent(
   overlay: O,
-  environment: ViewEnvironment
+  environment: ViewEnvironment,
+  turnOffChrome: Boolean = true,
+  setContent: (ComponentDialog, ScreenViewHolder<C>) -> Unit = { d, v -> d.setContentView(v.view) }
 ): OverlayDialogHolder<O> {
   // Note that we always tell Android to make the window non-modal, regardless of our own
   // notion of its modality. Even a modal dialog should only block events within
-  // the appropriate bounds, but Android makes them block everywhere.
+  // the appropriate bounds, but Android makes them block everywhere by default.
```

Comment:
It's not impossible, just a bit inconvenient. e.g. they can always turn the flag back on in their custom `setContentView` method. 

I definitely want to minimize the chance of clobbering this bit by accident. Without it we lose the main promise of `BodyAndOverlaysScreen`, that it will restrict the windows to the area of the body view.

It occurs to me that `toggleChrome` and `setContent` are redundant. I'm inclined to get rid of parameter, make the function public, and have the default `setContent` implementation call that. That's more like how the deprecated `abstract class ScreenOverlayDialogFactory` works

Reply:
Renamed `Dialog.turnOffChrome()` function to `Dialog.fixBackgroundAndDimming().